task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -90,13 +142,31 @@ impl MgsHandler {
         if self.common.sp_update.is_preparing() {
             Some(sys_get_timer().now + 1)
         } else {
-            None
+            self.locked
+                .iter()
+                .flat_map(|k| match k {
+                    LockState::UnlockedUntil(t) => Some(*t),
+                    _ => None,
+                })
+                .min()
```

Comment:
I did a double-take here because `None < Some(_)`, but just checking I got it: this is `flat_map`, not `map`, so all the `None`s are discarded and we're only taking the min of the `Some(*t)`s?

Reply:
Added a comment in f6b3a5b8a17a to explain what it's doing 😄